sql/core/src/test/resources/sql-tests/inputs/udf/postgreSQL/udf-join.sql

```diff
@@ -8,6 +8,9 @@
 --
 -- This test file was converted from postgreSQL/join.sql.
 
+-- Disable BroadcastHashJoin optimization to avoid changing result order when we enable AQE
+--SET spark.sql.autoBroadcastJoinThreshold = -1
```

Comment:
Disable BHJ optimization for the whole `SQLQueryTestSuite` would cause other tests to have different result order and fail. So, I think we could only disable the failed sql query itself instead to reduce the scope, which fix the test as well.